Editor/Windows/AssetPaletteWindow.cs

```diff
@@ -151,18 +151,35 @@ private void UpdateMouseOverStates()
                                             DragAndDrop.objectReferences.Length > 0 &&
                                             DragAndDrop.GetGenericData(EntryDragGenericDataType) == null;
         }
-        
+
         private void PerformKeyboardShortcuts()
         {
             if (Event.current.type != EventType.KeyDown)
                 return;
-            
-            if (Event.current.keyCode == KeyCode.Return && IsRenaming)
+
+            if (IsRenaming)
             {
-                StopAllRenames();
-                return;
+                switch (Event.current.keyCode)
+                {
+                    case KeyCode.Escape:
+                    case KeyCode.Return:
+                    case KeyCode.KeypadEnter:
+                        // Using a delay call here to allow canceling of the rename
+                        EditorApplication.delayCall += StopAllRenames;
```

Comment:
Does this mean escape counts as confirming the rename?

Unity behaviour is that pressing escape cancels a rename without applying it.

Also, what is the delay call for exactly?

Reply:
Escape cancels a rename without applying it in this case, that's what the delay call allows. Without the delay call it will apply either way!